core/commons/che-core-commons-lang/src/test/java/org/eclipse/che/commons/lang/TopologicalSortTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.commons.lang;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.testng.Assert.assertEquals;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class TopologicalSortTest {
+
+  @Test(dataProvider = "sortingCases")
+  public void shouldApplySorting(List<Var> list, List<Var> expectedSorting) throws Exception {
+    // when
+    sort(list);
+
+    // then
+    assertEquals(list, expectedSorting);
+  }
+
+  @DataProvider
+  public static Object[][] sortingCases() {
+    return new Object[][] {
+      // keep order if no dependencies
+      new Object[] {
+        vars(var('b', "."), var('a', "."), var('c', ".")),
+        vars(var('b', "."), var('a', "."), var('c', "."))
+      },
+
+      // sort dependents after dependencies
+      new Object[] {
+        vars(var('a', "b"), var('b', "."), var('c', ".")),
+        vars(var('b', "."), var('a', "b"), var('c', "."))
+      },
+
+      // sort dependents after dependencies, multiple dependencies
+      new Object[] {
+        vars(var('a', "bc"), var('b', "."), var('c', ".")),
+        vars(var('b', "."), var('c', "."), var('a', "bc"))
+      },
+
+      // sort dependents after dependencies, check dependee moved after dependencies
+      new Object[] {
+        vars(var('d', "."), var('a', "c"), var('b', "."), var('c', ".")),
+        vars(var('d', "."), var('b', "."), var('c', "."), var('a', "c"))
+      },
+
+      // test the robustness against cycles
+      new Object[] {vars(var('a', "a")), vars(var('a', "a"))},
```

Comment:
Just one more small proposal for cases names
```java
// item depends on itself
// cycle with 2 items
// cycle with 3 items + 1 independent item
```
The last case is quite tricky not sure if I can easily clear formulate description.

Reply:
explanations added, hopefully they're intelligible :)